pkg/controller/seed-controller-manager/kubernetes/pending.go

```diff
@@ -43,6 +43,11 @@ func (r *Reconciler) reconcileCluster(ctx context.Context, cluster *kubermaticv1
 		return nil, err
 	}
 
+	// Apply etcdLauncher flag
+	if err := r.ensureEtcdLauncherFeatureFlag(ctx, cluster); err != nil {
```

Comment:
I think this is defaulting, and should be done at admission phase. At the moment we don't have an admission webhook for `Cluster` resources yet, but I think this is the way to go long term. This is not a blocker as what you did is consistent with the approach used today.

Reply:
I agree.